benchmarks/foveation/main.cpp

```diff
@@ -0,0 +1,771 @@
+}
+void FoveationBenchmarkApp::UpdateRenderShaderParams()
+{
+    ShaderParams params    = {};
+    params.seed            = 0u;
+    params.noiseWeights    = float3(1.0f, 1.0f, 1.0f);
+    params.color           = float3(1.0f, 0.0f, 1.0f);
+    params.extraHashRounds = mRender.extraHashRounds;
+    if (mRender.renderPattern == RenderPattern::DYNAMIC_NOISE) {
+        params.seed = static_cast<uint32_t>(GetFrameCount());
+    }
+    else if (mRender.renderPattern == RenderPattern::CONSTANT) {
+        params.noiseWeights = float3(0.001f, 0.001f, 0.001f);
+    }
+
+    void* pData = nullptr;
+    PPX_CHECKED_CALL(mRender.uniformBuffer->MapMemory(0, &pData));
+    memcpy(pData, &params, sizeof(ShaderParams));
+    mRender.uniformBuffer->UnmapMemory();
+}
+
+void FoveationBenchmarkApp::RecordPostCommands(uint32_t imageIndex)
+{
+    // Wait for and reset image acquired fence
+    PPX_CHECKED_CALL(mSync.imageAcquiredFence->WaitAndReset());
+
+    // Build render command buffer
+    PPX_CHECKED_CALL(mPost.cmd->Begin());
+    {
+        mPost.cmd->TransitionImageLayout(mRender.drawPass->GetRenderTargetTexture(0), PPX_ALL_SUBRESOURCES, grfx::RESOURCE_STATE_RENDER_TARGET, grfx::RESOURCE_STATE_PIXEL_SHADER_RESOURCE);
+        mPost.cmd->BeginRenderPass(mPost.drawPass);
+        {
+            mPost.cmd->SetScissors(mPost.drawPass->GetScissor());
+            mPost.cmd->SetViewports(mPost.drawPass->GetViewport());
+            mPost.cmd->Draw(mPost.fullscreenQuad, 1, &mPost.descriptorSet);
+        }
+        mPost.cmd->EndRenderPass();
+        mPost.cmd->TransitionImageLayout(mRender.drawPass->GetRenderTargetTexture(0), PPX_ALL_SUBRESOURCES, grfx::RESOURCE_STATE_PIXEL_SHADER_RESOURCE, grfx::RESOURCE_STATE_RENDER_TARGET);
+
+        grfx::ImagePtr postImage      = mPost.drawPass->GetRenderTargetTexture(0)->GetImage();
+        grfx::ImagePtr swapchainImage = GetSwapchain()->GetColorImage(imageIndex);
+
+        mPost.cmd->TransitionImageLayout(postImage, PPX_ALL_SUBRESOURCES, grfx::RESOURCE_STATE_RENDER_TARGET, grfx::RESOURCE_STATE_COPY_SRC);
+        mPost.cmd->TransitionImageLayout(swapchainImage, PPX_ALL_SUBRESOURCES, grfx::RESOURCE_STATE_PRESENT, grfx::RESOURCE_STATE_COPY_DST);
+
+        grfx::ImageBlitInfo blitInfo = {};
+        blitInfo.srcImage.offsets[1] = {postImage->GetWidth(), postImage->GetHeight(), 1};
+        blitInfo.dstImage.offsets[1] = {swapchainImage->GetWidth(), swapchainImage->GetHeight(), 1};
+
+        mPost.cmd->BlitImage(&blitInfo, postImage, swapchainImage);
+
+        mPost.cmd->TransitionImageLayout(postImage, PPX_ALL_SUBRESOURCES, grfx::RESOURCE_STATE_COPY_SRC, grfx::RESOURCE_STATE_RENDER_TARGET);
+        mPost.cmd->TransitionImageLayout(swapchainImage, PPX_ALL_SUBRESOURCES, grfx::RESOURCE_STATE_COPY_DST, grfx::RESOURCE_STATE_PRESENT);
+    }
+    PPX_CHECKED_CALL(mPost.cmd->End());
+}
+
+void FoveationBenchmarkApp::SaveImage(grfx::ImagePtr image, const std::string& filepath, grfx::ResourceState resourceState) const
```

Comment:
This seems to duplicate most of the code in Application::TakeScreenshot. I understand we are saving a different non-swapchain image? If that's the case, can we move this function to Application and then re-use it here and in Application::TakeScreenshot?